### ISSUE TITLE
Need to dynamically change header background color while scrolling

### ISSUE THREAD

**Issue by ablbol** [react-community/react-navigation#3108]
Hi, I have a Stack Navigator Screen with header.  I would like to be able to change the header's backgroundColor while scrolling the screen.  I am using Animated API for that.  Thinks work fine if I use my own custom header but will received warning when using the default React Navigation header.   This is what I am doing. In the screen I have the following code:

```
  constructor(props) {
    super(props);
    this.scrollY = new Animated.Value(0);
  }

  componentWillMount() {
    this.props.navigation.setParams({
      bgColor: this.scrollY.interpolate({
        inputRange: [0, SCREEN_HEIGHT / 2 - 40],
        outputRange: ['transparent', 'rgb(255,0,0)'],
        extrapolate: 'clamp',
      }),
    });
  }

  render() {
    return (
      <ScrollView
        style={{ backgroundColor: 'black' }}
        contentContainerStyle={{ flexGrow: 1 }}
        onScroll={Animated.event([
          { nativeEvent: { contentOffset: { y: this.scrollY } } },
        ])}
        scrollEventThrottle={16}
      >
        <View>
.....................

```

I also have navigationOptions that should get the bgColor animated value from params.  Here is the code.  I commented the custom header that works fine.  

```
      navigationOptions: props => {
        debugger;
        return {
          tabBarVisible: false,
          headerStyle: {
            position: 'absolute',
            top: 0,
            left: 0,
            right: 0,
            backgroundColor: !props.navigation.state.params
              ? 'transparent'
              : props.navigation.state.params.bgColor,
            borderBottomWidth: 0,
            elevation: 0,
          },

          //   header: props => {
          //     debugger;
          //     return (
          //       <Animated.View
          //         style={{
          //           position: 'absolute',
          //           top: 0,
          //           left: 0,
          //           right: 0,
          //           height: 50,
          //           padding: 20,
          //           backgroundColor: !props.navigation.state.params
          //             ? 'transparent'
          //             : props.navigation.state.params.bgColor,
          //           borderBottomWidth: 0,
          //           elevation: 0,
          //         }}
          //       >
          //         <Text>this is header</Text>
          //       </Animated.View>
          //     );
          //   },
        };
      },
```
And this is the warning that I am getting.
Please let me know what I can do to fix this problem.

![1](https://user-images.githubusercontent.com/22548671/33823294-0a54827e-de74-11e7-8253-4020c95e42ca.png)
![2](https://user-images.githubusercontent.com/22548671/33823303-11423ffe-de74-11e7-9b1d-45bcf45530ec.png)

| software         | version
| ---------------- | -------
| react-navigation | "^1.0.0-beta.21"
| react-native     | "0.50.3"
| node             | v8.3.0
| npm or yarn      | yarn

**Comment by kelset** [comment]
Well - the warning says it, the `backgroungColor` property you are passing is invalid.

My bet is that it is related to what's written in the "note" [here in the doc](https://reactnavigation.org/docs/intro/headers#Header-interaction-with-screen-component) after the code snippet.

**Comment by ablbol** [comment]
Hi kelset,
Thanks for your help.
I checked the note that you referred to.  I am already following the note.  I am setting the backgroundColor to transparent when the params is not set yet.  I have a feeling the problem has something to do with a View in the header that should be animated.  By the way, if you look at second orange screen, which is a continuation of the warning, you will see that it refers to SafeAreaView and AnimatedComponent.

**Comment by kelset** [comment]
No, you check for `props.navigation.state.params` instead of checking for `props.navigation.state.params. bgColor`. I think that's a key difference.

**Comment by ablbol** [comment]
I changed the navigationOptions to be safer but still the same warning.
```
      navigationOptions: ({ navigation }) => {
        const { params = {} } = navigation.state;
        return {
          tabBarVisible: false,
          headerStyle: {
            position: 'absolute',
            top: 0,
            left: 0,
            right: 0,
            // backgroundColor: 'transparent',
            backgroundColor: params.bgColor ? params.bgColor : 'transparent',
            borderBottomWidth: 0,
            elevation: 0,
          },
        };
      },

```

**Comment by kelset** [comment]
I think you should also try to console.log the bgColor value to see why it triggers the warning only once 🤔

That said, I'll ping @davepack which may know something more (but not sure he can).

**Comment by ablbol** [comment]
I tried to do a debug on the params constant which is this line:
```
        const { params = {} } = navigation.state;
```
This is what I got.
<img width="1531" alt="3" src="https://user-images.githubusercontent.com/22548671/33836853-116304c0-dea4-11e7-95cf-0ddd80396dee.png">

**Comment by kelset** [comment]
Maybe related/similar to this? https://github.com/facebook/react-native/issues/13044

**Comment by ablbol** [comment]
It is ok to set the backgroundColor of a view but not ok to set the color of a view so I don't see how these two things related!!

**Comment by SkrabakL** [comment]
Hi!
I did something like this couple of hours ago and replied to the #1 stack overflow question :D that comes up with react-navigation + native + custom header and animations. 

it might help a bit -> there seems to be a key part with the data binding ? 
Without preventing the component to render before the params where set I did not work for me, but after that it works flawlessly. Changing rgba colors in header. 

So give it a shot ;-) 

https://stackoverflow.com/questions/44928060/react-navigation-custom-header-animation/47803230#47803230
